scripts/tests/twister/test_handlers.py

```diff
@@ -729,6 +729,24 @@ def test_devicehandler_monitor_serial(
     )
 
 
+def test_devicehandler_monitor_serial_splitlines(mocked_instance):
+    halt_event = mock.Mock(is_set=mock.Mock(return_value=False))
+    ser = mock.Mock(
+        isOpen=mock.Mock(side_effect=[True, True, False]),
+        in_waiting=mock.Mock(return_value=False),
+        readline=mock.Mock(return_value='\nline1\nline2\n'.encode('utf-8'))
```

Comment:
can this test also reproduce the `\r\xff` case as well as `\r\n` ?

Reply:
See the comments above. While I can add this test, the benefit may be minimal.